### DISCUSSION TITLE
Has anyone implemented a custom checkout using Stripe Payment Element?

### DISCUSSION THREAD

**Discussion by DerekKHarris** [swellstores/community#116] [Help] [answered]
I've found examples in the Swell documents about using Stripe Card and separate elements but has anyone been able to integrate the Stripe Payment Element?

https://stripe.com/docs/payments/payment-element

**Comment by sags95** [comment, marked as answer]
Hey @DerekKHarris , you should be able to use the Stripe Payment Element (or other elements like the Payment Request Button), but you would have to implement this using stripe-js and some server-side code. 

For the server-side code, this will mainly be for creating the paymentIntents, Swell will handle capturing the paymentIntent when the order is submitted. It's important to create paymentIntents  with the `capture_method` and `setup_future_usage` fields.

```
const stripe = require('stripe')('sk_test_***');

const paymentIntent = await stripe.paymentIntents.create({
  amount: 1099,
  currency: 'usd',
  capture_method: manual,
  setup_future_usage: off-session
});
```

In the success callback for the Payment Element, you can then update the cart with the recently created paymentMethod and paymentIntent
```
swell.cart.update({
    billing: {
        intent: {
            stripe: {
                id: '<pi_>'
            }
        },
        card: {
            token:  '<pm_>'
        }
    }
})
 ```

In the case for Apple or Google Pay, you will see these appear as credit cards in the Swell dashboard.

**Reply by rinorzek** [reply]
@sags95 can you please help me with this too: https://github.com/orgs/swellstores/discussions/331
